packages/core/content-manager/admin/src/pages/ListView/components/BulkActions/PublishAction.tsx

```diff
@@ -333,15 +333,16 @@ const SelectedEntriesModalContent = ({
   const { publishMany: bulkPublishAction } = useDocumentActions();
   const [, { isLoading: isSubmittingForm }] = usePublishManyDocumentsMutation();
 
-  const selectedEntries = useTable('publishAction', (state) => state.selectedRows);
+  const selectedRows = useTable('publishAction', (state) => state.selectedRows);
 
-  const entriesToPublish = selectedEntries.reduce((acc, entry) => {
-    if (!validationErrors[entry.documentId]) {
-      acc.push(entry.documentId);
-    }
+  // Filter selected entries from the updated modal table rows
+  const selectedEntries = rows.filter((entry) =>
+    selectedRows.some((selectedEntry) => selectedEntry.documentId === entry.documentId)
+  );
 
-    return acc;
-  }, []);
+  const entriesToPublish = selectedEntries
+    .filter((entry) => !validationErrors[entry.documentId])
+    .map((entry) => entry.documentId);
```

Comment:
Do you know what the max number of entries this code will have to manage at once will be? Something like 10 or 20?

I'm asking because your change increases readability but is a bit worse for performance, so it's only worth it if it doesn't deal with lots of data

Reply:
its not for readability, previous change was done by me only but its throwing TS error for accumulators `documentId` type `never` not matching with selectedEntries.documentId type `string`. so I updated it with simpler code. I could use your help to resolve the issue.
For the performance as its just selectedEntries its going to be max the pagination size so its not really impacting performance alot IMO.